### PR TITLE
feat(share): #MAG-288 implement shared folder get and delete rules

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
- get shared folder when loading page
- can delete folder if it is mine only (with back-end security)
- folder and board share are deleted if the element is put in the bin (+ lightbox message to warn the user)
- deleted unused directive

## Checklist tests
- check if shared folder appears for user that is not owner
- try to delete said folder (should not be able to)
- delete my own shared folder and board --> should not be shared anymore when viewed in the bin

## Issue ticket number and link
[MAG-288](https://jira.support-ent.fr/browse/MAG-288)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java [hunk 1]
```diff
@@ -257,7 +257,7 @@ public Future<JsonObject> preDeleteBoards(String userId, List<String> boardIds,
         JsonObject query = new JsonObject()
                 .put(Field._ID, new JsonObject().put(Mongo.IN, new JsonArray(boardIds)))
                 .put(Field.OWNERID, userId);
-        JsonObject update = new JsonObject().put(Mongo.SET, new JsonObject().put(Field.DELETED, !restore));
+        JsonObject update = new JsonObject().put(Mongo.SET, new JsonObject().put(Field.DELETED, !restore)).put(Mongo.UNSET, new JsonObject().put(Field.SHARED, 1));
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java:260]: remove share when putting board in din

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 2]
```diff
@@ -38,14 +37,23 @@ public DefaultFolderService(String collection, MongoDb mongo, ServiceFactory ser
     @Override
     public Future<JsonArray> getFolders(UserInfos user, boolean isDeleted) {
         Promise<JsonArray> promise = Promise.promise();
-        QueryBuilder matcher = QueryBuilder.start(Field.OWNERID).is(user.getUserId());
+
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:40]: get shared folders

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 3]
```diff
@@ -171,13 +180,50 @@ private Future<List<String>> getBoardIdsInFolders(List<String> folderIds) {
...
+    }
+
 
     @Override
     public Future<JsonObject> preDeleteFoldersAndBoards(UserInfos user, List<String> folderIds, boolean restore) {
         Promise<JsonObject> promise = Promise.promise();
+        List<String> myFolderIds = new ArrayList<>();
 
-        this.preDeleteBoardsInFolderWithChildren(folderIds, restore, user.getUserId())
-                .compose(v -> this.preDeleteFoldersWithChildren(folderIds, restore, user.getUserId()))
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:180]: filter folder ids that are not mine

File: src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java [hunk 4]
```diff
@@ -215,7 +261,7 @@ private Future<JsonObject> preDeleteFolders(List<String> folderIds, boolean rest
         JsonObject query = new JsonObject()
                 .put(Field._ID, new JsonObject().put(Mongo.IN, new JsonArray(folderIds)))
                 .put(Field.OWNERID, ownerId);
-        JsonObject update = new JsonObject().put(Mongo.SET, new JsonObject().put(Field.DELETED, !restore));
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultFolderService.java:218]: remove share when putting folder in bin

**Review by flmartineau — APPROVED** [review]

**Review by Rseuret — APPROVED** [review]

**Pull request merged**